src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala

```diff
@@ -629,7 +658,30 @@ private[fastq] object FastqDemultiplexer {
                          comment: Option[String],
                          originalBases: Option[String] = None,
                          originalQuals: Option[String] = None,
-                         readInfo: Option[ReadInfo] = None)
+                         readInfo: Option[ReadInfo] = None,
+                         numAboveQualityThreshold: Int = 0) {
+
+    /** Masks bases that have a quality score less than to a specified threshold.
+      * @param threshold The threshold for masking bases, exclusive. Bases with a quality score < threshold will be masked
+      * @param qualityEncoding The encoding used for quality scores in the Fastq file.
+      * @return a new DemuxRecord with updated bases
+      */
+    def maskLowQualityBases(threshold: PhredScore, qualityEncoding: QualityEncoding, metricsThreshold: PhredScore): DemuxRecord = {
+
+      val quals = this.quals.toCharArray
+      val numAboveQualityThreshold = quals.filter(_ >= metricsThreshold).size
+
+      if (threshold <= 0 || this.quals.forall(_ >= threshold)) {
+        this.copy(numAboveQualityThreshold=numAboveQualityThreshold)
+      } else {
+        val bases = this.bases.toCharArray
+        val maskedBases = bases.zip(quals).map {
+          case (base, qual) => if (qual < threshold) 'N' else base
+        }
```

Comment:
What happened to `for (i <- Range(0, this.quals.length)) if (quals.charAt(i).toByte < threshold) bases(i) = 'N'`?

Reply:
@jacarey didn't like it and suggested this instead

Reply:
@nh13 Is there a reason to side effect the bases array?

Reply:
@tfenne 

1. see [master](https://github.com/fulcrumgenomics/fgbio/blob/d7f4e79fe6314eee83140b4287aabbb20a0237ea/src/main/scala/com/fulcrumgenomics/fastq/DemuxFastqs.scala#L659)
2. see [@tfenne's comment on #716](https://github.com/fulcrumgenomics/fgbio/pull/716#discussion_r715172223)

Reply:
Fair enough. I'd be curious how much of a performance benefit this is